### ISSUE TITLE
remove py3.5 build because windows install fails with 1.7.3 build py_35_1

### ISSUE THREAD

**Issue by ReimarBauer** [Open-MSS/MSS#370]
**[Original report](https://bitbucket.org/wxmetvis/mss/issue/370) by Joern Ungermann (Bitbucket: [joernu76](https://bitbucket.org/joernu76), GitHub: [joernu76](https://github.com/joernu76)).**

----------------------------------------

mss crashes with an EntryPointError

```
#!python

Traceback (most recent call last):

  File "C:\Users\j.ungermann\Anaconda3\envs\mss-1.7.3-1\Scripts\mss-script.py",line 10, in <module>

    sys.exit(main())

  File "C:\Users\j.ungermann\Anaconda3\envs\mss-1.7.3-1\lib\site-packages\mslib\msui\mss_pyui.py", line 778, in main

    mainwindow.create_new_flight_track()

  File "C:\Users\j.ungermann\Anaconda3\envs\mss-1.7.3-1\lib\site-packages\mslib\msui\mss_pyui.py", line 548, in create_new_flight_track

    name="new flight track ({:d})".format(self.new_flight_track_counter))

  File "C:\Users\j.ungermann\Anaconda3\envs\mss-1.7.3-1\lib\site-packages\mslib\msui\flighttrack.py", line 157, in __init__

    self.load_settings()

  File "C:\Users\j.ungermann\Anaconda3\envs\mss-1.7.3-1\lib\site-packages\mslib\msui\flighttrack.py", line 172, in load_settings

    self.performance_settings = load_settings_pickle(self.settings_tag, DEFAULT_PERFORMANCE)

  File "C:\Users\j.ungermann\Anaconda3\envs\mss-1.7.3-1\lib\site-packages\mslib\utils.py", line 176, in load_settings_pickle

    _fs = open_fs(_dirname)

  File "C:\Users\j.ungermann\Anaconda3\envs\mss-1.7.3-1\lib\site-package\fs\opener\registry.py", line 177, in open_fs

    default_protocol=default_protocol,

  File "C:\Users\j.ungermann\Anaconda3\envs\mss-1.7.3-1\lib\site-package\fs\opener\registry.py", line 136, in open

    opener = self.get_opener(protocol)

  File "C:\Users\j.ungermann\Anaconda3\envs\mss-1.7.3-1\lib\site-packages\fs\opener\registry.py", line 86, in get_opener

    exception,

  File "<string>", line 3, in raise_from

fs.opener.errors.EntryPointError: could not load entry point; The 'typing~=3.6'

distribution was not found and is required by the application


```

The problem seems to be the requirement "typing~=3.6" in the requirements.txt file of the "fs" library. Any requirement for python system libraries fails in my Anaconda3 installation. These work on my linux Anaconda3 installation. 

Minimal testcase:

```
#!python

import pkg_resources
tst = pkg_resources.WorkingSet()
tst.require("typing")

```

**Comment by ReimarBauer** [comment]
**Original comment by Joern Ungermann (Bitbucket: [joernu76](https://bitbucket.org/joernu76), GitHub: [joernu76](https://github.com/joernu76)).**

----------------------------------------

Some more information. This is the list of packages it currently installs. For some reason it now uses python 3.5....

```
(mss-1.7.3-1) C:\Users\j.ungermann>conda list
# packages in environment at C:\Users\j.ungermann\Anaconda3\envs\mss-1.7.3-1:
#
# Name                    Version                   Build  Channel
apipkg                    1.5                      py35_0
appdirs                   1.4.3            py35h28b3542_0
argcomplete               1.9.4                    py35_0
asn1crypto                0.24.0                   py35_0
astropy                   3.0.4            py35hfa6e2cd_0
atomicwrites              1.2.1                    py35_0
attrs                     18.2.0           py35h28b3542_0
basemap                   1.1.0                    py35_2    conda-forge
bcrypt                    3.1.4            py35hfa6e2cd_0
beautifulsoup4            4.6.3                    py35_0
blas                      1.0                         mkl
ca-certificates           2018.03.07                    0
certifi                   2018.8.24                py35_1
cffi                      1.11.5           py35h74b6da3_1
cftime                    1.0.1            py35h452e1ab_0    conda-forge
chameleon                 3.4                      py35_0
chardet                   3.0.4                    py35_1
colorama                  0.3.9            py35h32a752f_0
cryptography              2.3              py35h74b6da3_0
cryptography-vectors      2.3.1                    py35_0
cycler                    0.10.0           py35hcc71164_0
execnet                   1.5.0                    py35_0
fs                        2.0.27                   py35_0    conda-forge
fs.sshfs                  0.8.0                    py35_0    conda-forge
fs.webdavfs               0.3.3                    py35_0    conda-forge
fs_filepicker             0.3.7                    py35_0    conda-forge
furl                      1.2.1                    py35_0
future                    0.16.0                   py35_2
geos                      3.5.1                    vc14_1  [vc14]  conda-forge
hdf4                      4.2.13                   vc14_0  [vc14]  conda-forge
html5lib                  1.0.1                    py35_0
humanfriendly             4.16.1                   py35_0
icc_rt                    2017.0.4             h97af966_0
icu                       58.2                     vc14_0  [vc14]  conda-forge
idna                      2.7                      py35_0
intel-openmp              2019.0                      118
jpeg                      9b                       vc14_2  [vc14]  conda-forge
jplephem                  2.8                pyh11f86e8_0    conda-forge
libpng                    1.6.34                   vc14_0  [vc14]  conda-forge
lxml                      3.8.0                    py35_0
matplotlib                2.0.2               np111py35_0
menuinst                  1.4.14           py35hfa6e2cd_0
mkl                       2019.0                      118
more-itertools            4.3.0                    py35_0
mss                       1.7.3                    py35_1    conda-forge
netcdf4                   1.2.2               np111py35_0
numpy                     1.11.3           py35h4a99626_4
openssl                   1.0.2o                   vc14_0  [vc14]  conda-forge
orderedmultidict          1.0                      py35_0
owslib                    0.16.0                   py35_0
paramiko                  2.4.1                    py35_0
paste                     2.0.3                    py35_0
pathlib2                  2.3.2                    py35_0
pillow                    2.9.0                    py35_0
pip                       10.0.1                   py35_0
pluggy                    0.7.1            py35h28b3542_0
proj4                     5.0.1                    vc14_0  [vc14]  conda-forge
psutil                    5.4.7            py35hfa6e2cd_0
py                        1.6.0                    py35_0
pyasn1                    0.4.4            py35h28b3542_0
pycparser                 2.18                     py35_1
pynacl                    1.2.1            py35h2fa13f4_0
pyopenssl                 18.0.0                   py35_0
pyparsing                 2.2.0                    py35_1
pyproj                    1.9.5.1                  py35_0
pyqt                      5.6.0            py35ha878b3d_6
pyshp                     1.2.12                   py35_0
pysocks                   1.6.8                    py35_0
pytest                    3.8.1                    py35_0
pytest-arraydiff          0.2              py35h39e3cac_0
pytest-astropy            0.4.0                    py35_0
pytest-doctestplus        0.1.3                    py35_0
pytest-openfiles          0.3.0                    py35_0
pytest-remotedata         0.3.0                    py35_0
python                    3.5.6                he025d50_0
python-dateutil           2.7.3                    py35_0
pytz                      2018.5                   py35_0
pywin32                   223              py35hfa6e2cd_1
qt                        5.6.2                    vc14_1  [vc14]  conda-forge
requests                  2.19.1                   py35_0
scipy                     1.1.0            py35hc28095f_0
setuptools                40.2.0                   py35_0
sgp4                      1.4                pyh24bf2e0_3    conda-forge
sip                       4.18.1           py35h6538335_2
six                       1.11.0                   py35_1
skyfield                  1.9                        py_0    conda-forge
tk                        8.5.19                   vc14_1  [vc14]  conda-forge
unicodecsv                0.14.1           py35h0d88516_0
urllib3                   1.23                     py35_0
vc                        14                   h0510ff6_3
vs2015_runtime            14.15.26706          h3a45250_0
webdavclient2             0.1.1                      py_0    conda-forge
webencodings              0.5.1                    py35_1
wheel                     0.31.1                   py35_0
win_inet_pton             1.0.1                    py35_1
wincertstore              0.2              py35hfebbdb8_0
zlib                      1.2.11                   vc14_0  [vc14]  conda-forge
```

**Comment by ReimarBauer** [comment]
**Original comment by Joern Ungermann (Bitbucket: [joernu76](https://bitbucket.org/joernu76), GitHub: [joernu76](https://github.com/joernu76)).**

----------------------------------------

Further exploration:
If I fix python to 3.6 (which install python 3.6.0) or python 3.6.6, the pillow DLL error raises it nasty head again.
Manually installing pillow from conda-forge resolves the issue.

Such, to install mss on Windows one needs to fix python to 3.6 as the fs/pkg_resources combination for python version for 3.5 seems to be broken on windows

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

conda-forge is planning on stopping new Python 3.5 builds by default after Oct. 1, 2018.  Please let us know now if this is problematic for you! This is consistent with our policy of building two versions of Python 3 (v3.6 and v3.7) and Python 2.7

https://twitter.com/condaforge/status/1045699331171528707

We have to move to 3.6.

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

At minmimum we may have to set a fixation pillow 5.*

https://anaconda.org/conda-forge/pillow

pillow      2.9.0 is rather old

**Comment by ReimarBauer** [comment]
**Original comment by Joern Ungermann (Bitbucket: [joernu76](https://bitbucket.org/joernu76), GitHub: [joernu76](https://github.com/joernu76)).**

----------------------------------------

I do not know, why conda installed py35 for the mss environment I newly created.

conda create -n env mss

used python 3.5 even though the root env uses python 3.6.

Only enforcing py36 with

conda create -n env mss python=3.6

used the more recent py3.6.0 (even though higher subversion are available).  I have no idead what is going on there.

With respect to pillow, the py36 environments use pillow 5.x, but fail due to the documented DLL problem, which can be fixed by using the conda-forge pillow.

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

Please check:

https://conda.io/docs/user-guide/configuration/use-condarc.html#disable-updating-of-dependencies-update-dependencies

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

Pillow issue described

https://github.com/python-pillow/Pillow/issues/2945

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

@joernu76 please file a bug report to:

https://github.com/ContinuumIO/anaconda-issues

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

We schould verify that  

pywin32-ctypes

is for py3.5 available and that it is installed by pillow

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

1.7.3-2 just merged, please retry

**Comment by ReimarBauer** [comment]
**Original comment by Joern Ungermann (Bitbucket: [joernu76](https://bitbucket.org/joernu76), GitHub: [joernu76](https://github.com/joernu76)).**

----------------------------------------

Success!? Now he installs by default py27 instead of py35....

**Comment by ReimarBauer** [comment]
**Original comment by Joern Ungermann (Bitbucket: [joernu76](https://bitbucket.org/joernu76), GitHub: [joernu76](https://github.com/joernu76)).**

----------------------------------------

So, in my Win7 virtual machine, mss works both in the py27 and in a py36 environment it created with "conda create -n mssenv mss python>=3".
So it seems fine. Kind of....

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

hmm,  this installs 3.6.6 python on linux. 

windows config or windows bug?

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

we skip python 3.5 from builds and set a minimum version 5 of pillow, fixes #370

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

Merged in ReimarBauer/mss/stable (pull request #588)

we skip python 3.5 from builds and set a minimum version 5 of pillow, fixes #370
